src/features/Map/ui/Map.tsx

```diff
@@ -1,28 +1,65 @@
-import React, { useEffect, useRef } from "react";
-import { YaMap, Geocoder } from "react-native-yamap";
+import React, { useEffect, useRef, useState } from "react";
+import { YaMap } from "react-native-yamap";
 import { styles } from "./Map.styles";
 import { useSetlocationStore } from "@entities/user";
-import { useSetCurretLocationStore } from "@entities/user";
-import { Nullable } from "@shared/api";
+import { Nullable, ParkingInf, ParkingSchema } from "@shared/api";
+import { ParkingMarker } from "@features/ParkingMarker";
 
-export const Map = () => {
-    const { currentLocation } = useSetCurretLocationStore();
+interface Map {
+    isPositionNeed: boolean;
+    parkingData: ParkingSchema[];
+    setIsModalVisible?: (isModalVisible: boolean) => void;
+    setParkingInf?: (parkingInf: ParkingInf) => void;
+}
 
+export const Map: React.FC<Map> = ({ isPositionNeed, parkingData, setIsModalVisible, setParkingInf }) => {
+    const [lastMarkerClickTimestamp, setLastMarkerClickTimestamp] = useState<number>(Date.now());
+    const [mapReady, setMapReady] = useState<boolean>(false);
+    const [markers, setMarkers] = useState<ParkingSchema[]>();
     const { location } = useSetlocationStore();
-
     const mapRef = useRef<Nullable<YaMap>>(null);
 
-    useEffect(() => {
-        if (mapRef.current && currentLocation) {
-            mapRef.current.setCenter({ lat: currentLocation.lat, lon: currentLocation.lon }, 10, 0, 0);
-        }
-    }, [currentLocation]);
-
     useEffect(() => {
         if (mapRef.current && location) {
             mapRef.current.setCenter({ lat: location.lat, lon: location.lon }, 10, 0, 0);
         }
-    }, [location]);
+    }, [location, mapReady]);
+
+    useEffect(() => {
+        setMarkers(parkingData);
+    }, [parkingData]);
+
+    const handlepPressOnMarker = (value: ParkingSchema) => {
```

Comment:
make this func as a prop to map to handle onPress outside of the Map component. Btw you've made a mistake
in func name